include/boost/numeric/ublas/tensor/tensor.hpp

```diff
@@ -38,798 +37,605 @@ namespace boost { namespace numeric { namespace ublas {
+    inline
+    const_iterator end () const {
+        return data_.end();
+    }
+
+    /// \brief return an iterator after the last element of the basic_tensor
+    inline
+    const_iterator cend () const {
+        return data_.cend ();
+    }
+
+    /// \brief Return an iterator on the first element of the basic_tensor
+    inline
+    iterator begin () {
+        return data_.begin();
+    }
+
+    /// \brief Return an iterator at the end of the basic_tensor
+    inline
+    iterator end () {
+        return data_.end();
+    }
+
+    /// \brief Return a const reverse iterator before the first element of the reversed basic_tensor (i.e. end() of normal basic_tensor)
+    inline
+    const_reverse_iterator rbegin () const {
+        return data_.rbegin();
+    }
+
+    /// \brief Return a const reverse iterator before the first element of the reversed basic_tensor (i.e. end() of normal basic_tensor)
+    inline
+    const_reverse_iterator crbegin () const {
+        return data_.crbegin();
+    }
+
+    /// \brief Return a const reverse iterator on the end of the reverse basic_tensor (i.e. first element of the normal basic_tensor)
+    inline
+    const_reverse_iterator rend () const {
+        return data_.rend();
+    }
+
+    /// \brief Return a const reverse iterator on the end of the reverse basic_tensor (i.e. first element of the normal basic_tensor)
+    inline
+    const_reverse_iterator crend () const {
+        return data_.crend();
+    }
+
+    /// \brief Return a const reverse iterator before the first element of the reversed basic_tensor (i.e. end() of normal basic_tensor)
+    inline
+    reverse_iterator rbegin () {
+        return data_.rbegin();
+    }
+
+    /// \brief Return a const reverse iterator on the end of the reverse basic_tensor (i.e. first element of the normal basic_tensor)
+    inline
+    reverse_iterator rend () {
+        return data_.rend();
+    }
 
-	inline
-	void resize( [[maybe_unused]] size_type sz ){
-		if constexpr( detail::is_resizable_v<array_type>){
-			data_.resize(sz);
-		}
-	}
+private:
 
 #if 0
-	// -------------
-	// Serialization
-	// -------------
-
-	/// Serialize a tensor into and archive as defined in Boost
-	/// \param ar Archive object. Can be a flat file, an XML file or any other stream
-	/// \param file_version Optional file version (not yet used)
-	template<class Archive>
-	void serialize(Archive & ar, const unsigned int /* file_version */){
-		ar & serialization::make_nvp("data",data_);
-	}
+    // -------------
+    // Serialization
+    // -------------
+
+    /// Serialize a basic_tensor into and archive as defined in Boost
+    /// \param ar Archive object. Can be a flat file, an XML file or any other stream
+    /// \param file_version Optional file version (not yet used)
+    template<class Archive>
+    void serialize(Archive & ar, const unsigned int /* file_version */){
+        ar & serialization::make_nvp("data",data_);
+    }
```

Comment:
I recommend to delete this serialization function.
It should be first defined as a project.

Reply:
Ok.